<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LoopParam</class>
 <widget class="QWidget" name="LoopParam">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>937</width>
    <height>697</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>环路参数</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resonrces.qrc">
    <normaloff>:/images/app.ico</normaloff>:/images/app.ico</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true">QWidget{
	color:#fff;
	/*background-color: rgb(240,240,240);*/
	background-color: rgb(24,28,40);
}
QLabel{
	background-color: rgb(41,45,57);
}

QPushButton{
	color:white;
	border:2px solid #147e3a;
	background-color:#019b48;
	width:50px;
	height:30px;
}

QPushButton:hover{
	background-color:#019b48;
	font-size:12px;
	font-weight:bold;
}

QLabel{
	color:white;
}

QGroupBox{
	border:2px solid rgb(41,45,57);
	color:white;
	background-color: rgb(41,45,57);
	border-radius:4px;
}

QLineEdit{	
	background-color: rgb(255, 255, 255);
	border:1px solid #147e3a;
	width:100px;
	height:30px;
	color:black;
}
QComboBox{
	background-color: rgb(255, 255, 255);
	width:100px;
	height:27px;
	border-radius:4px;
	border:1px solid rgb(41,45,57);
	color:black;
}

QLineEdit:hover{
	border:1px solid rgb(0,149,255);
}

QTabWidget:pane{
	border-top:1px solid rgb(240, 240, 240);
	/*border:1px solid rgb(41,45,57);*/
}

QTabBar::tab{
	color:white;
	width:50px;
	height:25px;
	background-color:#019b48;
	margin-right: 5px;
	margin-bottom:-6px;
}

QTabBar::tab:selected{
	background-color:rgb(0,149,255);
	font-weight:bold;
	height:28px;
	border:1.5px solid rgb(0,110,255);
}

QMessageBox{
	background-color:rgb(41,45,57);
}

QMessageBox QPushButton:enabled{
	border-radius: 5px;
	width: 50px;
	height: 20px;
	background-color: #019b48;
	color: white;
	font-size: 12px;
}
</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="2,4">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout" stretch="3,3,3,2">
     <property name="spacing">
      <number>3</number>
     </property>
     <item>
      <widget class="QGroupBox" name="currentLoopGroupBox">
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_8" stretch="1,0,1">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>3</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>3</number>
        </property>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_9">
          <item>
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>电流环</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line_5">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="styleSheet">
             <string notr="true">background-color: rgb(255, 255, 255);</string>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="gridGroupBox_2">
          <property name="styleSheet">
           <string notr="true">QLineEdit{
	border:1px solid rgb(41,45,57);
	height:25px;
	width:100px;
	border-radius:5px;
}
QLineEdit:hover{
	border:1px solid rgb(0,149,255);
}</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_3" columnstretch="2,2,2,2,1">
           <property name="leftMargin">
            <number>20</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <property name="horizontalSpacing">
            <number>6</number>
           </property>
           <property name="verticalSpacing">
            <number>4</number>
           </property>
           <item row="1" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>I值</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="currentPValue"/>
           </item>
           <item row="0" column="2">
            <widget class="QLabel" name="label_35">
             <property name="text">
              <string>电流类型</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="currentIValue"/>
           </item>
           <item row="1" column="2">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>电流幅值</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>P值</string>
             </property>
            </widget>
           </item>
           <item row="1" column="3">
            <widget class="QLineEdit" name="currentamplitudeLineEdit"/>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>D值</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLineEdit" name="currentDValue"/>
           </item>
           <item row="2" column="2">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>正弦频率</string>
             </property>
            </widget>
           </item>
           <item row="2" column="3">
            <widget class="QLineEdit" name="sinusoidalfrequencyLineEdit"/>
           </item>
           <item row="1" column="4">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>A</string>
             </property>
            </widget>
           </item>
           <item row="2" column="4">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Hz</string>
             </property>
            </widget>
           </item>
           <item row="0" column="3">
            <widget class="QComboBox" name="currentTypeComboBox">
             <item>
              <property name="text">
               <string>阶跃电流</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_24" stretch="4,2,2">
          <property name="rightMargin">
           <number>10</number>
          </property>
          <item>
           <spacer name="horizontalSpacer_11">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="currentLoopTestBtn">
            <property name="text">
             <string>测试</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="currentTestDataPlotBtn">
            <property name="text">
             <string>测试图形显示</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="SpeedLoopGroupBox">
       <property name="styleSheet">
        <string notr="true">QGroupBox{
	border:2px solid rgb(41,45,57);
	color:white;
	background-color: rgb(41,45,57);
	border-radius:4px;
}</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_14" stretch="1,0,1">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>3</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>3</number>
        </property>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_15">
          <item>
           <widget class="QLabel" name="label_15">
            <property name="text">
             <string>速度环</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line_8">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="styleSheet">
             <string notr="true">background-color: rgb(255, 255, 255);</string>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="gridGroupBox_5">
          <property name="styleSheet">
           <string notr="true">QLineEdit{
	border:1px solid rgb(41,45,57);
	height:25px;
	width:100px;
	border-radius:5px;
}
QLineEdit:hover{
	border:1px solid rgb(0,149,255);
}</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_6" columnstretch="2,2,2,2,1">
           <property name="leftMargin">
            <number>20</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <property name="verticalSpacing">
            <number>4</number>
           </property>
           <item row="0" column="0">
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>P值</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLineEdit" name="speedLoopDValue"/>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="speedLoopPValue"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>I值</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_16">
             <property name="text">
              <string>D值</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="speedLoopIValue"/>
           </item>
           <item row="0" column="2">
            <widget class="QLabel" name="label_40">
             <property name="text">
              <string>设定速度</string>
             </property>
            </widget>
           </item>
           <item row="0" column="3">
            <widget class="QLineEdit" name="setSpeedLineEdit"/>
           </item>
           <item row="0" column="4">
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>RPM</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_27" stretch="4,2,2">
          <property name="rightMargin">
           <number>10</number>
          </property>
          <item>
           <spacer name="horizontalSpacer_14">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="speedLoopTestBtn">
            <property name="text">
             <string>测试</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="velocityTestDataPlotBtn">
            <property name="text">
             <string>测试图形显示</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="positionLoopGroupBox">
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_16" stretch="1,0,1">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>3</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>3</number>
        </property>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_17">
          <item>
           <widget class="QLabel" name="label_20">
            <property name="text">
             <string>位置环</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line_9">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="styleSheet">
             <string notr="true">background-color: rgb(255, 255, 255);</string>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="gridGroupBox_6">
          <property name="styleSheet">
           <string notr="true">QLineEdit{
	border:1px solid rgb(41,45,57);
	height:25px;
	width:100px;
	border-radius:5px;
}
QLineEdit:hover{
	border:1px solid rgb(0,149,255);
}</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_7" columnstretch="2,2,2,2,1">
           <property name="leftMargin">
            <number>20</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <property name="verticalSpacing">
            <number>4</number>
           </property>
           <item row="1" column="0">
            <widget class="QLabel" name="label_11">
             <property name="text">
              <string>I值</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="positionLoopPValue"/>
           </item>
           <item row="0" column="2">
            <widget class="QLabel" name="label_41">
             <property name="text">
              <string>最大速度</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="positionLoopIValue"/>
           </item>
           <item row="1" column="2">
            <widget class="QLabel" name="label_21">
             <property name="text">
              <string>目标位置</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_22">
             <property name="text">
              <string>P值</string>
             </property>
            </widget>
           </item>
           <item row="1" column="3">
            <widget class="QLineEdit" name="targetPositionLineEdit"/>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_23">
             <property name="text">
              <string>D值</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLineEdit" name="positionLoopDValue"/>
           </item>
           <item row="1" column="4">
            <widget class="QLabel" name="label_25">
             <property name="text">
              <string>度</string>
             </property>
            </widget>
           </item>
           <item row="0" column="4">
            <widget class="QLabel" name="label_13">
             <property name="text">
              <string>RPM</string>
             </property>
            </widget>
           </item>
           <item row="0" column="3">
            <widget class="QLineEdit" name="MaxSpeedLineEdit"/>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_28" stretch="4,2,2">
          <property name="rightMargin">
           <number>10</number>
          </property>
          <item>
           <spacer name="horizontalSpacer_15">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="positionLoopTestBtn">
            <property name="text">
             <string>测试</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="positionTestDataPlotBtn">
            <property name="text">
             <string>测试图形显示</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="settingGroupBox">
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_12" stretch="1,0,1">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>3</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>3</number>
        </property>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_13">
          <item>
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>设置</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line_7">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="styleSheet">
             <string notr="true">background-color: rgb(255, 255, 255);</string>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QGroupBox" name="gridGroupBox_4">
          <property name="styleSheet">
           <string notr="true">QLineEdit{
	border:1px solid rgb(41,45,57);
	height:25px;
	width:100px;
	border-radius:5px;
}
QLineEdit:hover{
	border:1px solid rgb(0,149,255);
}</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_5" columnstretch="1,5,1">
           <property name="leftMargin">
            <number>61</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>97</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <property name="horizontalSpacing">
            <number>8</number>
           </property>
           <property name="verticalSpacing">
            <number>5</number>
           </property>
           <item row="0" column="2">
            <widget class="QLabel" name="label_38">
             <property name="text">
              <string>P/S^2</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_39">
             <property name="text">
              <string>加速度</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="accelerationLineEdit"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_17">
             <property name="text">
              <string>减速度</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="decelerationLineEdit"/>
           </item>
           <item row="1" column="2">
            <widget class="QLabel" name="label_42">
             <property name="text">
              <string>P/S^2</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_26" stretch="1,1,1,1">
          <item>
           <spacer name="horizontalSpacer_13">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="backBtn">
            <property name="text">
             <string>上一步</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="nextStepBtn">
            <property name="text">
             <string>下一步</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="verticalGroupBox_2">
     <layout class="QVBoxLayout" name="verticalLayout_2" stretch="5,0,5,1">
      <item>
       <widget class="WaveformWidget" name="waveformwidget1" native="true">
        <property name="styleSheet">
         <string notr="true">background-color: rgb(41,45,57);</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line">
        <property name="styleSheet">
         <string notr="true">background-color: rgb(255, 255, 255);</string>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="WaveformWidget" name="waveformwidget2" native="true">
        <property name="styleSheet">
         <string notr="true">background-color: rgb(41,45,57);</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout" stretch="6,9,9,9,9,6">
        <property name="spacing">
         <number>9</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="clearWaveBtn">
          <property name="text">
           <string>清除波形</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_3">
          <property name="text">
           <string>保存电流环数据</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_2">
          <property name="text">
           <string>保存速度环数据</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton">
          <property name="text">
           <string>保存位置环数据</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>WaveformWidget</class>
   <extends>QWidget</extends>
   <header>waveformwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resonrces.qrc"/>
 </resources>
 <connections/>
</ui>
